### ISSUE TITLE
Implement a fast reading mode.

### ISSUE THREAD

**Issue by MicahGale** [idaholab/MontePy#382]
**Is your feature request related to a problem? Please describe.**
A common use case of MontePy is model interrogation, where you are interested in just getting a few pieces of data from say one cell.
For large models this isn't very practical as loading the whole model can take upwards of a minute, and makes prototyping the code hard. 

**Describe the solution you'd like**
I proposed making a mode to read the input file in which only the objects the user cares about are loaded.

For instance `montepy.fast_read(cells={1,2,3})`

would only read cells 1,2,3 and their supporting surfaces, materials, transforms, modifiers, etc.

This does present opportunities for doing bad ideas like trying to modify a partially loaded model.

This would require significant work under the hood, and would need to really optimize the process of identifying relevant objects.

**Comment by tjlaboss** [comment]
A  "fast-read, read-only" mode would be extremely useful.

**Comment by MicahGale** [comment]
What do you mean by read-only here?

**Comment by tjlaboss** [comment]
Because...

> This does present opportunities for doing bad ideas like trying to modify a partially loaded model.

...then, having a partially loaded model that the user cannot modify makes sense. It might be [too annoying to make read-only versions](https://stackoverflow.com/questions/19022868/how-to-make-dictionary-read-only) of objects.

**Comment by MicahGale** [comment]
Yeah... I think just a lot of warnings in the documentation is the way to go.

**Comment by tjlaboss** [comment]
Idea from our meeting with @dodu94: on-the-fly parsing

**Comment by MicahGale** [comment]
@tjlaboss is there any reason to keep this separate from #529 given how effective JIT parsing will be?

**Comment by tjlaboss** [comment]
No, JIT parsing effectively accomplishes this. You can close it now, or with #529.

**Comment by MicahGale** [comment]
Yep I'm closing this to combine it with #529.
